blocks/search/search.js

```diff
@@ -104,172 +100,128 @@ function addSearchByListeners(wrapper, form) {
 function populateFilter(select, items) {
   const docRange = document.createRange();
   let htmlFragment = '';
+  const urlParams = new URLSearchParams(window.location.search);
   items.forEach((item) => {
+    const itemValue = urlParams.get('cat') ? item.toLowerCase() : item;
     htmlFragment += `
-      <option value="${item.toLowerCase()}">${item}</option>
+      <option value="${itemValue}">${item}</option>
     `;
   });
   const fragment = docRange.createContextualFragment(htmlFragment);
   select.appendChild(fragment);
 }
 
+export const getAndApplySearchResults = async ({ isFirstSet }) => {
+  const { MAX_PRODUCTS_PER_QUERY } = SEARCH_CONFIG;
+  const urlParams = new URLSearchParams(window.location.search);
+  const resultsSection = document.querySelector('.results-list__section');
+  const resultsList = document.querySelector('.results-list__list');
+  const query = urlParams.get('q');
+  const offsetParam = urlParams.get('offset');
+  const make = urlParams.get('make') === 'null' ? undefined : urlParams.get('make');
+  const model = urlParams.get('model') === 'null' ? undefined : urlParams.get('model');
+  const searchType = urlParams.get('st');
+  const category = urlParams.get('category');
+  const targetOffset = isFirstSet && offsetParam === '0' ? 0 : parseInt(offsetParam) + 1;
+  if (!isFirstSet) {
+    const newUrl = new URL(window.location);
+    newUrl.searchParams.set('offset', targetOffset);
+    window.history.pushState({ path: newUrl.href }, '', newUrl.href);
+  }
+  const filtersWrapper = document.querySelector('.filters-wrapper');
+  const loadingLabel = getTextLabel('loading_label');
+  let loadingElement = document.querySelector('.loading');
+  if (!loadingElement) {
+    loadingElement = createElement('div', { classes: 'loading' });
+    resultsSection.append(loadingElement);
+  }
+  loadingElement.textContent = loadingLabel;
+  const offset = targetOffset * parseInt(MAX_PRODUCTS_PER_QUERY);
+  const searchParams = { query, offset, make, model, searchType, category };
+  const { results, categories } = await fetchSearchResults(searchParams);
+  loadingElement.remove();
+  const searchResultsSection = document.querySelector('.search-results-section');
+  const titleElement = searchResultsSection.querySelector('.title');
+  if (results?.length > 0) {
+    results.forEach((result) => {
+      const liElement = productCard(result, searchType);
+      resultsList.appendChild(liElement);
+    });
+    const titleContent = getTextLabel('search_results_title');
+    const type = searchType === 'cross' ? 'cross-reference' : 'parts';
+    const titleText = `${titleContent} ${searchType === 'cross' ? `${type}: "${query}"` : `${make || ''} ${model || ''} ${query} ${type}`}`;
+    titleElement.textContent = titleText;
+    const buttonTextContent = getTextLabel('pagination_button');
+    const resultsCountElement = document.querySelector('.displayed-text');
```

Comment:
Can this selector be improved? `displayed-text`
I assume this could come from various components

Reply:
Also bellow you are accessing the innerText of `resultsCountElement`.
If resultsCountElement is null...

Reply:
I renamed it to reflect the purpose more obviously. Validated `resultsCountElement` too